File: server/internal/server/objects/units/Skills/ISkill.go
package Skills

import (
	"server/internal/server"
	"server/internal/server/objects"
	"server/pkg/packets"
)

type ISkill interface {
	Type() packets.SkillType
	Name() string
	Initiate(UnitId int32, client *server.Client, enemyClient *server.Client, gameService GameSkillService)
	ValidTarget(pos objects.Vector2I) bool
	Use(skillId int32, pos objects.Vector2I)
	SendPacket(packet packets.Msg)
}

type Data struct {
	Cooldown    int
	AP          int
	Name        string
	Description string
	Type        packets.SkillType
	Range       int
}

type AttackData struct {
	Data
	Damage int
}

File: Client/Objects/Units/Skill/ISkill.cs
using Godot;
using Packets;


public interface ISkill
{
    public IUnit Unit { get; set; }
    public string Name();
    public Node3D Node { get; set; }
    public void Init(IUnit unit, Node3D node, SkillData data);
    public SkillType Type();
    public void Use(Vector2I position);
    public void SendPacket(Vector2I position, int skillId);
}